modin/numpy/math.py

```diff
@@ -348,12 +362,48 @@ def mean(x1, axis=None, dtype=None, out=None, keepdims=None, *, where=True):
 # Maximum and minimum are ufunc's in NumPy, which means that our array's __array_ufunc__
 # implementation will automatically handle this, so we can just use NumPy's maximum/minimum
 # since that will route to our array's ufunc.
-maximum = numpy.maximum
+def maximum(
+    x1, x2, out=None, where=True, casting="same_kind", order="K", dtype=None, subok=True
+):
+    x1 = try_convert_from_interoperable_type(x1)
+    if not isinstance(x1, array):
+        ErrorMessage.single_warning(
+            f"Modin NumPy only supports objects of modin.numpy.array types for maximum, not {type(x1)}. Defaulting to NumPy."
+        )
+    return numpy.maximum(
+        x1,
+        x2,
+        out=out,
+        where=where,
+        casting=casting,
+        order=order,
+        dtype=dtype,
+        subok=subok,
+    )
+
 
-minimum = numpy.minimum
+def minimum(
+    x1, x2, out=None, where=True, casting="same_kind", order="K", dtype=None, subok=True
+):
+    x1 = try_convert_from_interoperable_type(x1)
```

Comment:
```suggestion
    x1 = try_convert_from_interoperable_type(x1)
    x2 = try_convert_from_interoperable_type(x2)
```

Reply:
We don't need to convert the second operand, since we do so inside the array class!

Reply:
Apparently I do not fully understand how the code works.

Could you explain where we convert the second operand?
And why can't we convert the first operand there?

At least the next check can be rewritten as follows:
```python
if not isinstance(x1, (array, pd.DataFrame, pd.Series)):
```

Reply:
So all of the `modin.numpy` methods are simply wrappers that internally call `modin.numpy.array.<method>`. E.g.:
```python
def add(...):
x1 = try_convert_from_interoperable_type(x1)
if isinstance(x1, array):
x1.add ...
```
So we need to make sure that the first operand is converted from a modin pandas object to a modin numpy object so that it can call the appropriate method, but inside the `modin.numpy.array.add` method, the first line attempts to convert the other operand to an array.

Take `__add__` as an example. On [this](https://github.com/RehanSD/modin/blob/a94b84cd6ebc63a0d951476ee1fd34629ae7c6de/modin/numpy/arr.py#L868) line, we call `_binary_op` - `_binary_op`, makes sure that the second operand is of type array, and errors if it isn't on these lines: https://github.com/RehanSD/modin/blob/a94b84cd6ebc63a0d951476ee1fd34629ae7c6de/modin/numpy/arr.py#L634-L638 - so we don't need to convert the second operand at the numpy layer.

Reply:
Got it! Thank you for the clarification.